firedrake/interpolation.py

```diff
@@ -617,6 +617,9 @@ def interpolate(
             self.point_eval_interpolator.interpolate(
                 f_src_at_src_node_coords, transpose=True, output=output
             )
+            # FIXME we cannot yet use actual cofunctions, see the note above
+            if not isinstance(output, firedrake.Cofunction):
+                output = output.riesz_representation("l2")
```

Comment:
This looks like a bug.

Reply:
```
            f_src_at_dest_node_coords_dest_mesh_decomp = firedrake.Cofunction(
                self.to_input_ordering_interpolator.V.dual()
            )
```

Yes, I spotted this in #2297.

Reply:
It should now be dual safe. @nbouziani does this look good?